Golang client vendor issue

## Problem
When building the go application using a vendor directory, the build fails due to the vendor command not copying the headers:

## How to reproduce

1. In any go application using the latest golang client, run `go mod vendor`
2. build or run the application

## Extra context

### Here is the output on my machine
```bash
$ go run main.go  
# github.com/tigerbeetle/tigerbeetle-go/pkg/types
vendor/github.com/tigerbeetle/tigerbeetle-go/pkg/types/bindings.go:9:10: fatal error: ../native/tb_client.h: No such file or directory
    9 | #include "../native/tb_client.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

Doing so without the vendor directory (or with `-mod=mod` flag) will solve the issue but vendor directory is not used anymore.

### Here is my go.mod

```go.mod
module tbtest

go 1.24.4

require github.com/tigerbeetle/tigerbeetle-go v0.16.53 // indirect
```

--- a/build.zig
+++ b/build.zig
@@ -5,6 +5,7 @@ const Query = std.Target.Query;
 const Mode = std.builtin.Mode;
 
 const config = @import("./src/config.zig");
+const stdx = @import("./src/stdx/stdx.zig");
 
 const VoprStateMachine = enum { testing, accounting };
 const VoprLog = enum { short, full };
@@ -1266,7 +1267,7 @@ fn build_go_client(
         // We don't need the linux-gnu builds.
         if (comptime std.mem.indexOf(u8, platform[0], "linux-gnu") != null) continue;
 
-        const name = if (comptime std.mem.eql(u8, platform[0], "x86_64-linux-musl"))
+        const platform_name = if (comptime std.mem.eql(u8, platform[0], "x86_64-linux-musl"))
             "x86_64-linux"
         else if (comptime std.mem.eql(u8, platform[0], "aarch64-linux-musl"))
             "aarch64-linux"
@@ -1274,7 +1275,7 @@ fn build_go_client(
             platform[0];
 
         const query = Query.parse(.{
-            .arch_os_abi = name,
+            .arch_os_abi = platform_name,
             .cpu_features = platform[2],
         }) catch unreachable;
         const resolved_target = b.resolveTargetQuery(query);
@@ -1298,10 +1299,20 @@ fn build_go_client(
         lib.bundle_compiler_rt = true;
         lib.step.dependOn(&bindings.step);
 
+        assert(std.mem.count(u8, lib.out_lib_filename, ".") == 1);
+        const file_name: []const u8, const extension: []const u8 = stdx.cut(
+            lib.out_lib_filename,
+            ".",
+        ).?;
+
         // NB: New way to do lib.setOutputDir(). The ../ is important to escape zig-cache/.
         step_clients_go.dependOn(&b.addInstallFile(
             lib.getEmittedBin(),
-            b.pathJoin(&.{ "../src/clients/go/pkg/native/", name, lib.out_filename }),
+            std.fmt.allocPrint(b.allocator, "../src/clients/go/pkg/native/{s}_{s}.{s}", .{
+                file_name,
+                platform_name,
+                extension,
+            }) catch @panic("OOM"),
         ).step);
     }
 }

--- a/build.zig
+++ b/build.zig
@@ -5,7 +5,6 @@ const Query = std.Target.Query;
 const Mode = std.builtin.Mode;
 
 const config = @import("./src/config.zig");
-const stdx = @import("./src/stdx/stdx.zig");
 
 const VoprStateMachine = enum { testing, accounting };
 const VoprLog = enum { short, full };
@@ -1299,20 +1298,21 @@ fn build_go_client(
         lib.bundle_compiler_rt = true;
         lib.step.dependOn(&bindings.step);
 
-        assert(std.mem.count(u8, lib.out_lib_filename, ".") == 1);
-        const file_name: []const u8, const extension: []const u8 = stdx.cut(
-            lib.out_lib_filename,
-            ".",
-        ).?;
+        const file_name: []const u8, const extension: []const u8 = cut: {
+            assert(std.mem.count(u8, lib.out_lib_filename, ".") == 1);
+            var it = std.mem.splitScalar(u8, lib.out_lib_filename, '.');
+            defer assert(it.next() == null);
+            break :cut .{ it.next().?, it.next().? };
+        };
 
         // NB: New way to do lib.setOutputDir(). The ../ is important to escape zig-cache/.
         step_clients_go.dependOn(&b.addInstallFile(
             lib.getEmittedBin(),
-            std.fmt.allocPrint(b.allocator, "../src/clients/go/pkg/native/{s}_{s}.{s}", .{
+            b.fmt("../src/clients/go/pkg/native/{s}_{s}.{s}", .{
                 file_name,
                 platform_name,
                 extension,
-            }) catch @panic("OOM"),
+            }),
         ).step);
     }
 }

--- a/src/clients/go/.gitignore
+++ b/src/clients/go/.gitignore
@@ -8,8 +8,5 @@ zigcc*
 main
 main.exe
 
-pkg/native/aarch64-linux/
-pkg/native/aarch64-macos/
-pkg/native/x86_64-linux/
-pkg/native/x86_64-macos/
-pkg/native/x86_64-windows/
+pkg/native/*.a
+pkg/native/*.lib

--- a/src/clients/go/pkg/native/native.go
+++ b/src/clients/go/pkg/native/native.go
@@ -1,0 +1,4 @@
+// Adds reference to sub-folders containing the external (non-Go) files
+// required to build the TigerBeetle client. Otherwise the `tb_client.h`
+// header and library object files would be pruned during `go mod vendor`.
+package native

--- a/src/clients/go/tb_client.go
+++ b/src/clients/go/tb_client.go
@@ -2,11 +2,11 @@ package tigerbeetle_go
 
 /*
 #cgo CFLAGS: -g -Wall
-#cgo darwin,arm64 LDFLAGS: ${SRCDIR}/pkg/native/aarch64-macos/libtb_client.a -ldl -lm
-#cgo darwin,amd64 LDFLAGS: ${SRCDIR}/pkg/native/x86_64-macos/libtb_client.a -ldl -lm
-#cgo linux,arm64 LDFLAGS: ${SRCDIR}/pkg/native/aarch64-linux/libtb_client.a -ldl -lm
-#cgo linux,amd64 LDFLAGS: ${SRCDIR}/pkg/native/x86_64-linux/libtb_client.a -ldl -lm
-#cgo windows,amd64 LDFLAGS: -L${SRCDIR}/pkg/native/x86_64-windows -ltb_client -lws2_32 -lntdll
+#cgo darwin,arm64 LDFLAGS: ${SRCDIR}/pkg/native/libtb_client_aarch64-macos.a -ldl -lm
+#cgo darwin,amd64 LDFLAGS: ${SRCDIR}/pkg/native/libtb_client_x86_64-macos.a -ldl -lm
+#cgo linux,arm64 LDFLAGS: ${SRCDIR}/pkg/native/libtb_client_aarch64-linux.a -ldl -lm
+#cgo linux,amd64 LDFLAGS: ${SRCDIR}/pkg/native/libtb_client_x86_64-linux.a -ldl -lm
+#cgo windows,amd64 LDFLAGS: -L${SRCDIR}/pkg/native -ltb_client_x86_64-windows -lws2_32 -lntdll
 
 #include <stdlib.h>
 #include <string.h>
@@ -34,6 +34,7 @@ import (
 	"unsafe"
 
 	"github.com/tigerbeetle/tigerbeetle-go/pkg/errors"
+	_ "github.com/tigerbeetle/tigerbeetle-go/pkg/native"
 	"github.com/tigerbeetle/tigerbeetle-go/pkg/types"
 )
 

